### PR TITLE
Add Linux support

### PR DESCRIPTION
**Pull request by lefterisgar**
Hi there! My friend and I noticed that this wonderful app lacks support for GNU/Linux.

We have created a custom bash script which checks and installs all the required dependencies and then builds the application from source. It checks if Python 3 and pip are installed. It can also automatically install pip for most Linux distros and if there is one we don't support (yet) it falls back to an alternative method which is distro-agnostic. Then it checks and installs the project dependencies as well. After that it proceeds to build the package while also actively monitoring the logs for any errors. Whether the build has succeeded or failed, it prints a message. If it has failed, it asks you if you want to see the log. The dependency checks only happen during the first run to avoid slowing down the build process after everything is setup. If the build fails for any reason (except if it has been aborted manually), the checks run again to ensure everything is working as it should. We have also updated the README.md to reflect all those changes.

Dependencies have been bumped as well, as this results in better performance (at least on Linux).
The end result is a binary file which runs on every distribution we have tested. Check our fork if you want to download and test it.

The script in action:
Screenshot_20220608_181025

We hope you will find it useful and it integrates well with the project!
Thanks!

### CODE REVIEW

**Comment by bradytheinventor** [comment]
This is very cool and actually comes at the perfect time - I just set up an Ubuntu VM to look into building for Linux.
Let me review this and then I will merge it.

**Comment by lefterisgar** [comment]
That's great! You don't need to do anything manually. Just run the script and it should work. If it doesn't, let us know!

**Review by bradytheinventor — APPROVED** [review]
Left some comments for my own benefit after this gets merged. I will make some edits but the script works well as-is.

I wonder if this script solves the distro problem - pyinstaller builds a binary for the platform you run it on, which is not guaranteed to run on any arbitrary Linux distro. Linux users are often technically capable, so maybe just letting them build their own binaries (with a Debian-compatible prebuilt as a fallback) would be the best solution.

Anyway, if you have thoughts or additional changes let me know! Otherwise I'll merge this tomorrow.

File: build.sh
Comment (lefterisgar) [reply build.sh:128]: I think that the binary file should work on all popular distributions as that is the case most of the time. I can't imagine anyone trying to run it on an obscure distro from 2013. Right?

- Yes, I think it does. At least from what I can see there is only one file in build/ regardless of how many times I have run the script.
- The initial idea was to pipe everything to a file to make the output look cleaner. Is your suggestion to make everything visible or only the errors? It could point the user to just open the log file in case of any error.
- It is a good idea to change the directory.

File: build.sh
Comment (lefterisgar) [reply build.sh:115]: Initially, that is how we had implemented it. However, it was easier to put everything script-related to a separate directory which is easier to remove. The tmp directory is also used when wget downloads pip. If the build fails then the tmp directory is easily removed and normal operation is restored.

File: README.md
Comment (lefterisgar) [reply README.md:17]: Great idea indeed. I'm going to see how it can be included in the script as well (for ease of use).

**Comment by lefterisgar** [comment]
I'm going to try implementing some of the changes you have suggested. Yeah, it should be good tomorrow because I'm doing them right now.

File: build.sh
Comment (GEROGIANNIS) [reply build.sh:128]: On Manjaro which i am on the binary works perfectly!

**Comment by lefterisgar** [comment]
I have made a couple of improvements. Let me know if you are okay with that.

**Comment by lefterisgar** [comment]
If the build has failed:
Screenshot_20220609_175610

File: build.sh
Comment (bradytheinventor) [inline build.sh:128]: Consider:
```
--distpath bin \
2>&1 | tee build/latest.log
```

File: build.sh
Comment (bradytheinventor) [inline build.sh:136]: I think the user will get more value if they can see the log from pyinstaller in-console. That way they know pyinstaller is making progress, see errors in real-time, etc.

I would recommend sending stderr to stdout, then piping that to `tee` so pyinstaller can log to the console and to a file. See above

File: build.sh
Comment (bradytheinventor) [inline build.sh:23]: privileges :)

**Review by bradytheinventor — APPROVED** [review]
I did an experiment: pyinstaller cleans `build/imd-gui` but not `build/`, so it's safe to store files there. Eventually I might move all the build scripts into `build/` so they aren't cluttering up the top-level directory, but I think it's also ok the way it is now.

I'll wait until tomorrow again to merge this, in case you have comments or final edits.

File: build.sh
Comment (lefterisgar) [reply build.sh:23]: Fixed. My bad!

File: build.sh
Comment (lefterisgar) [reply build.sh:136]: Taken care of.

**Comment by lefterisgar** [comment]
Great! Now everything should be in proper order. I applied all the fixes you suggested and also added some comments. I also took the opportunity to fix the YouTube link in the README. If you don't like it just revert the commit.

I think that you can merge it now.

**Pull request merged**
